website/src/server/index.tsx

```diff
@@ -74,6 +74,8 @@ if (process.env.COMPRESS_ASSETS === 'true') {
 app.use(sw());
 
 if (process.env.NODE_ENV === 'production' || process.env.NODE_ENV === 'test') {
+  // Always try to send static files ASAP. For files that don't exist,
+  // the server will server-side render the client router and set the appropriate status code.
   app.use(mount('/dist', serve(path.join(__dirname, '..', '..', 'dist'))));
```

Comment:
Add `no-cache` to all dist files, because of SW caching.

Reply:
`public, no-cache` is probably right for dist by default. You found an example of a dist file whose URL stays constant even when the file contents change.

Eventually for the dist files whose filenames are guaranteed to change whenever the file contents change, `public, max-age=31536000, immutable`, would be optimal.

Reply:
How about this approach (currently implemented in this PR). There are some other assets that also never change names, some of which are handled through some exclusion rules in webpack (not sure why).

To have a predictive rule on determining if the asset can be cached or not:
- Add a `.cached.` marker in the exported filename (mostly for chuncks that includes `[hash]`)
- Webpack worker plugin adds a `.worker.js` at the end for some of those files, but also these include `[hash]` and thus can be cached.

This would be an opt-in and shouldn't cause issues if we need to add more processing exclusion rules, they should not include the `.cached.` file segement.

<details><summary>These are all assets of website</summary>

```
dist/assets/018c0c569c00c0e707dcc30d1f40b354.png
dist/assets/1f423990e6bfdae184f966c86dc22c36.png
dist/assets/27b3706e74e4ee9f45ea34967a2f3fe4.png
dist/assets/48505df7e1a1eb25606c4789c3a3e235.png
dist/assets/52001a8b742a278688f6b516f931d81b.png
dist/assets/54da9e6c3dac08e684c23ebb5a3d2248.png
dist/assets/58b405f0e890b2e0c10e2881e13ae570.png
dist/assets/6b544ca2f9529bc2b22aa6978ebd5862.svg
dist/assets/6dcf149da3a69d5ee3fe7fb6933f5fa6.png
dist/assets/7bc60d33ce267a372e7d7b9807495282.png
dist/assets/7f658e7f03b868a6e934914996462329.png
dist/assets/9682e34e1f058feab75a232e3b9f33c6.png
dist/assets/af05a3786092b27e789607af7f5ab349.png
dist/assets/bc351fd24f9bd32bc131f122d42c1a77.svg
dist/assets/c51d82a43adccd86d66477bfc28dd39b.png
dist/assets/d14a681a76bdb229e305aa965b54dd18.png
dist/assets/d93a5d445d7b7a8be13f38362a9dcf7e.png
dist/assets/e392635154a17bd759d3bbb77e72eada.png
dist/assets/e95fe5167b7dfc17a242d6a9f10a60c3.png
dist/assets/ea07de06422d3379a4f579dfc4a779b9.png
dist/assets/f95e873b5154c5128ccb7bf69715156d.svg
dist/vendors~Editor/MonacoEditor.css
dist/vendors~Editor/MonacoEditor.css.map
dist/0.090843be60561cfc4f38.chunk.cached.worker.js
dist/0.1162b3ef4bc1c36fe5d4.chunk.cached.worker.js
dist/0.22e9a9be1201b9bd2641.chunk.cached.worker.js
dist/0.46d31f457e19cad8d352.chunk.cached.worker.js
dist/0.4dd0eddec8614de30d8c.chunk.cached.js
dist/0.7f09aa858bd49cbd2735.chunk.cached.worker.js
dist/1.4dd0eddec8614de30d8c.chunk.cached.js
dist/10.4dd0eddec8614de30d8c.chunk.cached.js
dist/12.4dd0eddec8614de30d8c.chunk.cached.js
dist/13.4dd0eddec8614de30d8c.chunk.cached.js
dist/14.4dd0eddec8614de30d8c.chunk.cached.js
dist/16.4dd0eddec8614de30d8c.chunk.cached.js
dist/17.4dd0eddec8614de30d8c.chunk.cached.js
dist/18.4dd0eddec8614de30d8c.chunk.cached.js
dist/19.4dd0eddec8614de30d8c.chunk.cached.js
dist/2.4dd0eddec8614de30d8c.chunk.cached.js
dist/20.4dd0eddec8614de30d8c.chunk.cached.js
dist/21.4dd0eddec8614de30d8c.chunk.cached.js
dist/22.4dd0eddec8614de30d8c.chunk.cached.js
dist/23.4dd0eddec8614de30d8c.chunk.cached.js
dist/24.4dd0eddec8614de30d8c.chunk.cached.js
dist/25.4dd0eddec8614de30d8c.chunk.cached.js
dist/26.4dd0eddec8614de30d8c.chunk.cached.js
dist/27.4dd0eddec8614de30d8c.chunk.cached.js
dist/28.4dd0eddec8614de30d8c.chunk.cached.js
dist/29.4dd0eddec8614de30d8c.chunk.cached.js
dist/3.4dd0eddec8614de30d8c.chunk.cached.js
dist/30.4dd0eddec8614de30d8c.chunk.cached.js
dist/31.4dd0eddec8614de30d8c.chunk.cached.js
dist/32.4dd0eddec8614de30d8c.chunk.cached.js
dist/33.4dd0eddec8614de30d8c.chunk.cached.js
dist/34.4dd0eddec8614de30d8c.chunk.cached.js
dist/35.4dd0eddec8614de30d8c.chunk.cached.js
dist/36.4dd0eddec8614de30d8c.chunk.cached.js
dist/37.4dd0eddec8614de30d8c.chunk.cached.js
dist/38.4dd0eddec8614de30d8c.chunk.cached.js
dist/39.4dd0eddec8614de30d8c.chunk.cached.js
dist/4.4dd0eddec8614de30d8c.chunk.cached.js
dist/40.4dd0eddec8614de30d8c.chunk.cached.js
dist/41.4dd0eddec8614de30d8c.chunk.cached.js
dist/42.4dd0eddec8614de30d8c.chunk.cached.js
dist/43.4dd0eddec8614de30d8c.chunk.cached.js
dist/44.4dd0eddec8614de30d8c.chunk.cached.js
dist/45.4dd0eddec8614de30d8c.chunk.cached.js
dist/46.4dd0eddec8614de30d8c.chunk.cached.js
dist/47.4dd0eddec8614de30d8c.chunk.cached.js
dist/48.4dd0eddec8614de30d8c.chunk.cached.js
dist/49.4dd0eddec8614de30d8c.chunk.cached.js
dist/5.4dd0eddec8614de30d8c.chunk.cached.js
dist/50.4dd0eddec8614de30d8c.chunk.cached.js
dist/51.4dd0eddec8614de30d8c.chunk.cached.js
dist/52.4dd0eddec8614de30d8c.chunk.cached.js
dist/53.4dd0eddec8614de30d8c.chunk.cached.js
dist/54.4dd0eddec8614de30d8c.chunk.cached.js
dist/55.4dd0eddec8614de30d8c.chunk.cached.js
dist/56.4dd0eddec8614de30d8c.chunk.cached.js
dist/57.4dd0eddec8614de30d8c.chunk.cached.js
dist/58.4dd0eddec8614de30d8c.chunk.cached.js
dist/59.4dd0eddec8614de30d8c.chunk.cached.js
dist/6.4dd0eddec8614de30d8c.chunk.cached.js
dist/60.4dd0eddec8614de30d8c.chunk.cached.js
dist/61.4dd0eddec8614de30d8c.chunk.cached.js
dist/62.4dd0eddec8614de30d8c.chunk.cached.js
dist/63.4dd0eddec8614de30d8c.chunk.cached.js
dist/64.4dd0eddec8614de30d8c.chunk.cached.js
dist/65.4dd0eddec8614de30d8c.chunk.cached.js
dist/66.4dd0eddec8614de30d8c.chunk.cached.js
dist/67.4dd0eddec8614de30d8c.chunk.cached.js
dist/68.4dd0eddec8614de30d8c.chunk.cached.js
dist/69.4dd0eddec8614de30d8c.chunk.cached.js
dist/7.4dd0eddec8614de30d8c.chunk.cached.js
dist/70.4dd0eddec8614de30d8c.chunk.cached.js
dist/71.4dd0eddec8614de30d8c.chunk.cached.js
dist/72.4dd0eddec8614de30d8c.chunk.cached.js
dist/73.4dd0eddec8614de30d8c.chunk.cached.js
dist/74.4dd0eddec8614de30d8c.chunk.cached.js
dist/75.4dd0eddec8614de30d8c.chunk.cached.js
dist/76.4dd0eddec8614de30d8c.chunk.cached.js
dist/77.4dd0eddec8614de30d8c.chunk.cached.js
dist/78.4dd0eddec8614de30d8c.chunk.cached.js
dist/79.4dd0eddec8614de30d8c.chunk.cached.js
dist/8.4dd0eddec8614de30d8c.chunk.cached.js
dist/80.4dd0eddec8614de30d8c.chunk.cached.js
dist/81.4dd0eddec8614de30d8c.chunk.cached.js
dist/82.4dd0eddec8614de30d8c.chunk.cached.js
dist/83.4dd0eddec8614de30d8c.chunk.cached.js
dist/84.4dd0eddec8614de30d8c.chunk.cached.js
dist/85.4dd0eddec8614de30d8c.chunk.cached.js
dist/86.4dd0eddec8614de30d8c.chunk.cached.js
dist/9.4dd0eddec8614de30d8c.chunk.cached.js
dist/EditorView.css
dist/EditorView.css.map
dist/Markdown/MarkdownPreview.css
dist/Markdown/MarkdownPreview.css.map
dist/app.bundle.js
dist/build-stats.js
dist/sw.bundle.js
```

</detauls>